src/ProtoBuf.FSharp/CodeGen.fs

```diff
@@ -117,6 +118,7 @@ let private emitFactory (resultType : Type) (zeroValuesPerField: ZeroValues.Fiel
         emitRecordDefault gen resultType
     | null -> // Is a type that isn't a record with no parameterless constructor. NOTE: This is significantly slower for deserialisation than alternative pathways.
         gen.Emit(OpCodes.Ldtoken, resultType)
+        gen.Emit(OpCodes.Call, MethodHelpers.getMethodInfo <@ Type.GetTypeFromHandle @> [| |])
```

Comment:
What does this line do? Why it is required?

Reply:
Ldtoken puts magical integral type id onto stack (as described in sections III.4.17 of [CLI standard](https://docs.microsoft.com/en-us/dotnet/fundamentals/standards)). It should first be converted to System.Type reference (methods in next line expect it), which `GetTypeFromHandle` does. I forgot to add it previously and some better question is how the heck it worked before .NET 6 (tests cover both places where it should be). Some weird runtime optimization, maybe?

Reply:
Thanks for the explanation.